common/persistence/sql/sqlplugin/sqlite/driver.go

```diff
@@ -35,11 +38,48 @@ import (
 )
 
 const (
-	goSqlDriverName       = "sqlite"
+	goSqlDriverName       = "sqlite_temporal"
 	sqlConstraintCodes    = sqlite3.SQLITE_CONSTRAINT | sqlite3.SQLITE_CONSTRAINT_PRIMARYKEY | sqlite3.SQLITE_CONSTRAINT_UNIQUE
 	sqlTableExistsPattern = "SQL logic error: table .* already exists \\(1\\)"
 )
 
+// Driver implements a SQL driver that returns a custom connection.
+// The custom connection provides ResetSession() and IsValid() methods,
+// preventing the SQL library from closing the connection when a transaction context is canceled.
+type Driver struct {
+	sqlite.Driver
+}
+
+func newDriver() driver.Driver {
+	return &Driver{sqlite.Driver{}}
+}
+
+func (d *Driver) Open(name string) (driver.Conn, error) {
+	c, err := d.Driver.Open(name)
+	if err != nil {
+		return nil, err
+	}
+	return &conn{c}, nil
+}
+
+type conn struct {
+	driver.Conn
+}
+
+// ResetSession does nothing.
+func (c *conn) ResetSession(ctx context.Context) error {
+	return nil
+}
+
+// IsValid always returns true. We only have one connection to the sqlite db. It should always be valid. Otherwise, we would have lost the database.
+func (c *conn) IsValid() bool {
+	return true
+}
```

Comment:
If the connection is not valid for some reason, we must have lost the database already. and if the connection is not valid, there is nothing we can do to create a new connection. Because of this I think it is safe to return valid in all cases.

Reply:
Put this in a comment in the code please.